### DISCUSSION TITLE
Testing DeleteAction with requiresConfirmation

### DISCUSSION THREAD

**Discussion by BradnDodd** [filamentphp/filament#13912] [Help]
### Package

Actions

### Package Version

3.2.97

### How can we help you?

Say for example I have a DeleteAction on a table that requires confirmation. How do I test that if you click the confirm of the DeleteAction modal that the associated model is deleted?. I have the test working if i disable confirmation but it fails to delete the model if I re-enable the confirmation modal

Action; 
```
ActionGroup::make([
                 DeleteAction::make()
                    ->requiresConfirmation(),
            ]),
```
Test;
```
it('can delete single post', function () {
    $post = Post::factory()->create();

    $livewire = livewire(ListPosts::class)
        ->callTableAction(
            \Filament\Tables\Actions\DeleteAction::class,
            $post
        );

    $this->assertModelMissing($post);
});
```

**Comment by leandrocfe** [comment]
`DeleteAction` has requiresConfirmation as default. Anyway, this shoud work.
What is the error?

**Comment by BradnDodd** [comment]
There is no error as such, the test fails as the model is not deleted

**Comment by rvzug** [comment]
Are you using SoftDeletes?
If so, below is not using the assertion you probably want to use, but it should work:

```php
it('can delete single post', function () {
    $post = Post::factory()->create();

    $livewire = livewire(ListPosts::class)
        ->callTableAction(
            \Filament\Tables\Actions\DeleteAction::class,
            $post
        );

    $post->refresh();

    $this->assertNotNull($post->deleted_at);
});
```

**Reply by BradnDodd** [reply]
It's not using SoftDeletes. I know it is isolated to the fact that the action is set to `->requiresConfirmation()` if I set ``->requiresConfirmation(false)` then the test passes. So that makes me know that my test is missing the clicking submit/ confirming the action but I can't find any documentation or obvious method when digging through the test stubs that would indicate the ability to confirm an action.

I tried doing the obvious and passing `['confirm' => true]` as an argument to `callTableAction()` but that didn't work

**Reply by rvzug** [reply]
Hmm. In previous tests that I've made (with ```->requiresConfirmation(true)```), tests pass when you call ```->callTableAction()```. But when the test fails, it should show you what went wrong: 'failed to assert xxx is xxx'.
